### PR TITLE
Fix loading values for virtual fields

### LINKED ISSUE #30562: Filtering on an aggregated or a custom expression field results in `GET` request to non-existent endpoint
### Describe the bug

If you try to filter on an aggregated or a custom expression value/dimension, it will result in a `GET` request to non-existent endpoint.

## Scenario 1 (custom expression)
1. New > Question > Raw data > People
2. Create custom column `[City]` and call it `CCity`
3. Click on Filter and choose `CCity`
4. See the error in the console

```
api.js:218
GET http://localhost:3000/api/field/expression%2CCCity%2C/values 404 (Not Found)

React_devtools_backend_compact.js:2367
Request entities,fields,expression,CCity,,values failed:
{ status: 404,
  data: 'API endpoint does not exist.',
  isCancelled: false
}

index.jsx:1
Uncaught (in promise)
{ status: 404,
  data: 'API endpoint does not exist.',
  isCancelled: false
}
```

## Scenario 2 (aggregation)
1. New > Question > Raw data > People
2. Summarize `Count` group by `Source`
3. Click on Filter and choose `Count`
4. See the error in the console

```
api.js:218
GET http://localhost:3000/api/field//values 400 (Bad Request)

React_devtools_backend_compact.js:2367
Request entities,fields,,values failed:
{
    "status": 400,
    "data": "<h1>Bad Message 400</h1><pre>reason: Ambiguous URI empty segment</pre>",
    "isCancelled": false
}

index.jsx:1
Uncaught (in promise)
{
    "status": 400,
    "data": "<h1>Bad Message 400</h1><pre>reason: Ambiguous URI empty segment</pre>",
    "isCancelled": false
}
```

### Expected behavior

If there is no field, don't call this endpoint.

### Logs

Included in the repro steps.

### Information about your Metabase installation

local dev, `master`, c43e85f866e65b8c92252b3f238d5fbfe63a6abf, H2, Sample Database

### Severity

P3 since it doesn't seem to affect the functionality or the UI in any way.

### Additional context

Tested all the way back to v43 and could see the same behavior.

### LINKED ISSUE #31662: Between filter causes an exception in the frontend on refresh
### Describe the bug

If you add a between filter with default values on a dashboard it will work on the first load, but on reload the FE will break

### To Reproduce

1) add a dashboard
2) add a number filter with "between" operator and add 2 values
3) save and refresh
4) if it doesn't break, try refreshing again

see
image

### Expected behavior

we shouldn't break the FE

### Logs

_No response_

### Information about your Metabase installation

```JSON
I think it only happens on master, but have my serious doubts
```

### Severity

P3?

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/30562
Fixes https://github.com/metabase/metabase/issues/31662
Reproduces https://github.com/metabase/metabase/issues/30562
Reproduces https://github.com/metabase/metabase/issues/31662

Previously the component tried to fetch field values for all fields, including ones that aren't backed by real DB fields. This PR fixes the issue and makes the component load data only for real fields.

How to verify:
- New -> Question -> Orders
- Add a custom column `[Total] + 1`
- Add a filter based on this column
- There should be no failed request for `values` as before

### CODE REVIEW

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js [hunk 1]
```diff
@@ -405,4 +411,15 @@ describe("FieldValuesWidget", () => {
       });
     });
   });
+
+  describe("custom expressions", () => {
+    const field = checkNotNull(metadata.field(EXPRESSION_FIELD_ID));
+
+    it("should not call fetchFieldValues", () => {
```
Comment (ranquild) [inline frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.unit.spec.js:418]: Follows the same pattern used in other tests in this file

File: frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx [hunk 2]
```diff
@@ -184,7 +184,11 @@ class FieldValuesWidgetInner extends Component {
   fetchFieldValues = async query => {
     if (query == null) {
       const { fields, fetchFieldValues } = this.props;
-      await Promise.all(fields.map(field => fetchFieldValues(field.id)));
+      await Promise.all(
+        fields
+          .filter(field => !field.isVirtual())
```
Comment (kamilmielnik) [inline frontend/src/metabase/components/FieldValuesWidget/FieldValuesWidget.jsx:189]: :x: 

I tried the branch locally and while reproducing it (just followed the steps) I noticed that there's a JS error coming from `dedupeValues` call right in the next line (192). It also happens afterwards, when trying to edit the filter.
The error is caused by this change.

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')
    at FieldValuesWidget.jsx:458:1
    at Array.map (<anonymous>)
    at dedupeValues (FieldValuesWidget.jsx:458:1)
    at _callee$ (FieldValuesWidget.jsx:184:1)
    at tryCatch (runtime.js:63:1)
    at Generator.invoke [as _invoke] (runtime.js:293:1)
    at Generator.next (runtime.js:118:1)
    at asyncGeneratorStep (index.jsx:1:1)
    at _next (index.jsx:1:1)
```

image

**Review by kamilmielnik — CHANGES_REQUESTED** [review]
This change breaks code that follows it

**Comment by ranquild** [comment]
@kamilmielnik fixed, please take a look

**Review by kamilmielnik — APPROVED** [review]
LGTM :+1:

**Pull request merged**